File: templates/Features/StoreNotificationsVB/Services/ActivationService_postaction.vb
﻿'{**
' These code blocks include the StoreNotificationsFeatureService Instance in the method `GetActivationHandlers()`
' and initializes it in the method `InitializeAsync()` in the ActivationService of your project.
'**}

'{[{
Imports Param_RootNamespace.Helpers
'}]}

Namespace Services
    Friend Class ActivationService
        Private Async Function StartupAsync() As Task
            '^^
            '{[{
            Await Singleton(Of StoreNotificationsFeatureService).Instance.InitializeAsync()
            '}]}
            Await Task.CompletedTask
        End Function

        Private Iterator Function GetActivationHandlers() As IEnumerable(Of ActivationHandler)
            '{[{
            yield Singleton(Of StoreNotificationsFeatureService).Instance
            '}]}
'{--{
            Exit Function'}--}
        End Function
    End Class
End Namespace

File: templates/Features/StoreNotifications/Services/ActivationService_postaction.cs
﻿//{**
// These code blocks include the StoreNotificationsFeatureService Instance in the method `GetActivationHandlers()`
// and initializes it in the method `InitializeAsync()` in the ActivationService of your project.
//**}

using System;
//{[{
using Param_RootNamespace.Helpers;
//}]}

namespace Param_ItemNamespace.Services
{
    internal class ActivationService
    {
        private async Task StartupAsync()
        {
            //^^
            //{[{
            await Singleton<StoreNotificationsFeatureService>.Instance.InitializeAsync();
            //}]}
            await Task.CompletedTask;
        }

        private IEnumerable<ActivationHandler> GetActivationHandlers()
        {
            //{[{
            yield return Singleton<StoreNotificationsFeatureService>.Instance;
            //}]}
//{--{
            yield break;//}--}
        }
    }
}
